src/internal/operators/switchMap.ts

```diff
@@ -123,13 +123,17 @@ class SwitchMapSubscriber<T, R> extends SimpleOuterSubscriber<T, R> {
       this.destination.error(error);
       return;
     }
-    const innerSubscription = this.innerSubscription;
+    let innerSubscription = this.innerSubscription;
     if (innerSubscription) {
       innerSubscription.unsubscribe();
     }
     const innerSubscriber = new SimpleInnerSubscriber(this);
     this.destination.add(innerSubscriber);
-    this.innerSubscription = innerSubscribe(result, innerSubscriber);
+    this.innerSubscription = innerSubscriber;
+    innerSubscription = innerSubscribe(result, innerSubscriber);
+    if (this.innerSubscription === innerSubscriber) {
```

Comment:
I think this is the same as the comment above. But please do check. I'm pretty sure that `innerSubscriber === innerSubscription`, and therefor also `=== this.innerSubscription`. So I think we just need to set `this.innerSubscription = innerSubscriber`, then call `innerSubscribe` and let the return value drop.

Reply:
Same as above, changed to just setting `this.innerSubscription = innerSubscriber`, and dropping the return value.